### ISSUE TITLE
make docs/index.html fails

### ISSUE THREAD

**Issue by tillmo** [spechub/Hets#1941]
```
make docs/index.html

Check log.haddock for results ...
haddock -o docs -h -s ../%F  \
            -t 'Hets - the Heterogeneous Tool Set' \
            -p Hets-Haddock-Prologue.txt --optghc=-DTAR_PACKAGE --optghc=-DNO_WGET --optghc=-DUNIX --optghc=-DUNI_PACKAGE --optghc=-DHASKELINE --optghc=-DHEXPAT --optghc=-DSERVER --optghc=-DWARP3 --optghc=-DHAXML --optghc=-DRDFLOGIC --optghc=-DCASLEXTENSIONS --optghc=-DGTKGLADE --optghc=-Wall --optghc=-fwarn-tabs --optghc=-fwarn-unrecognised-pragmas --optghc=-fno-warn-orphans --optghc=-fno-warn-unused-do-bind \
             OMDoc.hs COL.hs CspCASLProver.hs ExtModal.hs Proofs.hs CASL.hs Propositional.hs CMDL.hs Isabelle.hs CoCASL.hs s.hs TopHybrid.hs Hybrid.hs hets.hs CspCASL.hs GUI.hs Comorphisms.hs Logic.hs MMT.hs HasCASL.hs Modal.hs OWL2.hs Syntax.hs Taxonomy.hs Driver.hs ATC.hs Common.hs RelationalScheme.hs SoftFOL.hs Framework.hs Haskell.hs Static.hs CASL_DL.hs \
	>log.haddock 2>&1
Makefile:557: recipe for target 'docs/index.html' failed
make: *** [docs/index.html] Error 1
```

**Comment by cmaeder** [comment]
Hi Till, I don't know why you assigned this issue to me. I had a short look into the file `log.haddock` and I saw the error 

``` 
ATC/Id.hs:28:1: error:
    Could not find module ‘ATerm.Lib’
```

However, I was not able to install the package `aterm` (using cabal) due to

```
src/ATerm/AbstractSyntax.hs:59:36: error:
    Not in scope: type constructor or class ‘TypeRep’
```

Building the binaries using stack and creating docs using haddock do no longer fit together.
Cheers Christian

**Comment by cmaeder** [comment]
```
Persistence/Schema/OMSOrigin.hs:4:1: error:
    Failed to load interface for ‘Database.Persist.TH’
```

With a stack installed haddock I get the error above.

**Comment by tillmo** [comment]
Hi @cmaeder, many thanks for your help. I get a different error:
```
Persistence/Schema/OMSOrigin.hs:107:1: error:
    Parse error: naked expression at top level
    Perhaps you intended to use TemplateHaskell
Haddock coverage:
```

It seems that statements of form
```
derivePersistField "OMSOrigin"
```
in `Persistence/Schema/` cause this error. However, they compile with ghc. So the question is how to persuade haddock to behave like ghc here.

**Comment by cmaeder** [comment]
Like the compilation, the documentation should be generated using `stack haddock`. The file `Persistence/Schema/OMSOrigin.hs` has no header containing `LANGUAGE TemplateHaskell `.

**Comment by tillmo** [comment]
thanks for the hint. I corrected that, see #1942. Now I get
```
/home/till/projects/Hets/Persistence/SQLite.hs:10:1: error:
    Failed to load interface for ‘Control.Monad.Trans.Control’
    It is a member of the hidden package ‘monad-control-1.0.2.2’.
    Perhaps you need to add ‘monad-control’ to the build-depends in your .cabal file.
    Use -v to see a list of the files searched for.
```
Do you know how to fix this? Adding this in Hets.cabal has no influence on stack.yml.

**Comment by cmaeder** [comment]
Sorry, I can not reproduce this. I'm stuck with:

```
Persistence/Schema/OMSOrigin.hs:5:1: error:
    Failed to load interface for ‘Database.Persist.TH’
    Use -v to see a list of the files searched for.
```

And just `make` fails for me with:

```
glib > Preprocessing library glib-0.13.5.0...
glib > setup: Error in C header file.
glib >      
glib > /usr/include/glib-2.0/glib/gspawn.h:76: (column 22) [FATAL] 
glib >   >>> Syntax error!
glib >   The symbol `__attribute__' does not fit here.
```

**Comment by TpmKranz** [comment]
> And just `make` fails for me with:
> 
> ```
> glib > Preprocessing library glib-0.13.5.0...
> glib > setup: Error in C header file.
> glib >      
> glib > /usr/include/glib-2.0/glib/gspawn.h:76: (column 22) [FATAL] 
> glib >   >>> Syntax error!
> glib >   The symbol `__attribute__' does not fit here.
> ```

This should be fixed by #1939 once someone gets around to review and merge it.
